### PR TITLE
Fix fdr gene tagging when integers in var names/obsm columns

### PR DESCRIPTION
**Pull request by adamgayoso**
For more context see the discussion on discourse:

https://discourse.scverse.org/t/questions-about-running-differential-expression/1137/8

When using totalVI, if protein counts are in obsm as np array and no column names are provided, we generate sequential integers. This causes indexing issues in the fdr tagging as the index is mixed between gene names and the integers.

### CODE REVIEW

File: tests/models/test_models.py [hunk 1]
```diff
@@ -1238,37 +1238,24 @@ def test_totalvi(save_path):
...
     model.get_elbo(adata2)
 
     # test that we catch incorrect mappings
-    adata = synthetic_iid()
-    TOTALVI.setup_anndata(
-        adata,
-        batch_key="batch",
-        protein_expression_obsm_key="protein_expression",
-        protein_names_uns_key="protein_names",
-    )
```
Comment (adamgayoso) [inline tests/models/test_models.py:1258]: these were removed because they weren't doing anything

**Comment by adamgayoso** [comment]
I just want @PierreBoyeau to give it a quick look

**Review by PierreBoyeau — APPROVED** [review]
Looks good to me!

File: scvi/model/base/_utils.py [hunk 2]
```diff
@@ -279,15 +279,17 @@ def _de_core(
...
     cumulative_fdr = (1.0 - sorted_pgs).cumsum() / (1.0 + np.arange(len(sorted_pgs)))
     d = (cumulative_fdr <= fdr).sum()
-    pred_de_genes = sorted_genes[:d]
-    is_pred_de = np.zeros_like(cumulative_fdr).astype(bool)
-    is_pred_de[pred_de_genes] = True
+    is_pred_de = pd.Series(
+        np.zeros_like(cumulative_fdr).astype(bool), index=sorted_pgs.index
+    )
+    is_pred_de.iloc[:d] = True
+    is_pred_de = is_pred_de[original_index]
```
Comment (adamgayoso) [inline scvi/model/base/_utils.py:294]:
```suggestion
    is_pred_de = is_pred_de.loc[original_index]
```

**Pull request merged**